src/js/builtins/Bake.ts

```diff
@@ -130,11 +130,11 @@ export function renderRoutesForProdStatic(
         });
         if (paramGetter[Symbol.asyncIterator] != undefined) {
           for await (const params of paramGetter) {
-            callRouteGenerator(type, i, layouts, pageModule, params);
+            await callRouteGenerator(type, i, layouts, pageModule, params);
```

Comment:
This should check if it returned a Promise nad if that Promise is not resolved, and only then use await.